install/scripts/SetupExtendedTextInteractionConfigurationRegistry.php

```diff
@@ -0,0 +1,97 @@
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA;
+ *
+ * @author Sergei Mikhailov <sergei.mikhailov@taotesting.com>
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoQtiItem\install\scripts;
+
+use oat\oatbox\extension\script\ScriptAction;
+use oat\oatbox\reporting\Report;
+use oat\taoQtiItem\model\QtiCreator\ExtendedTextInteractionConfigurationRegistry;
+
+/**
+ * Run the following to enable Math Entry format option
+ * php index.php 'oat\taoQtiItem\install\scripts\SetupExtendedTextInteractionConfigurationRegistry' -m 1
+ *
+ * Run the following to disable Math Entry format option
+ * php index.php 'oat\taoQtiItem\install\scripts\SetupExtendedTextInteractionConfigurationRegistry' -m 0
+ */
+class SetupExtendedTextInteractionConfigurationRegistry extends ScriptAction
+{
+    public const HAS_MATH = 'math';
+
+    protected function provideOptions(): array
+    {
+        return [
+            self::HAS_MATH => [
+                'prefix'      => 'm',
+                'longPrefix'  => self::HAS_MATH,
+                'description' => 'Enables / disables Math entry on extended text interactions',
+            ],
+        ];
+    }
+
+    protected function provideDescription(): string
+    {
+        return sprintf('Sets `%s` values.', ExtendedTextInteractionConfigurationRegistry::class);
+    }
+
+    protected function run(): Report
+    {
+        $setValues = [];
+
+        /** @var ExtendedTextInteractionConfigurationRegistry $registry */
+        $registry = $this->propagate(new ExtendedTextInteractionConfigurationRegistry());
+
+        $this->setHasMath($registry, $setValues);
+
+        return $this->createReport($setValues);
+    }
+
+    private function createReport(array $setValues): Report
+    {
+        return $setValues
+            ? Report::createSuccess(
+                sprintf(
+                    "Applied the following configuration to `%s`\n%s",
+                    ExtendedTextInteractionConfigurationRegistry::class,
+                    json_encode($setValues)
+                )
+            )
+            : Report::createError(
+                sprintf('No values set to `%s`', ExtendedTextInteractionConfigurationRegistry::class)
+            );
+    }
+
+    private function setHasMath(ExtendedTextInteractionConfigurationRegistry $registry, array &$setValues): void
```

Comment:
This is very confusing method. Why is this created? Why don't just use `$this-getOption()` to retrieve this flag?

Reply:
Agreed that it is a bit confusing couldn't really invest more time into it, but this method, and potential future methods (in case more properties are added to the registry) are aimed to read and handle a single option of the command interface.

Reply:
I suggest returning to it when a new property would actually needed to be added to the registry.

Reply:
This looks like go lang method. You are using array reference instead using class attribute. **It's not wrong**. IMO it's overcomplicated and therefore less readable.

Reply:
And we are never going to return here again. This is almost certain.

Reply:
I don't see a reason to change it then to be fair.

Reply:
Found a better way of tracking the set properties.
Removed this method whatsoever.